libr/util/format.c

```diff
@@ -506,7 +506,8 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 	if (MUSTSET) {
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (MUSTSEE) {
-		char *timestr = strdup(asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc (26);
```

Comment:
malloc can fail

Reply:
@trufae Is there a safe malloc wrapper in radare2 (xmalloc, ect ...)

Reply:
We don't use those things AFAIK. Just check `if (timestr) { .... ; return from function }`

Reply:
Where is `26` coming from? Better to have a macro for it or something, instead of relying on an hardcoded number.

Reply:
man ascitime
```
      The asctime() function converts the broken-down time value tm  into  a
       null-terminated  string  with  the same format as ctime().  The return
       value points to a statically allocated string which might be overwrit‐
       ten  by  subsequent  calls to any of the date and time functions.  The
       asctime_r() function does the same, but stores the string in  a  user-
       supplied buffer which should have room for at least 26 bytes.
```

Reply:
Oh I see :/ Maybe define a macro in this file (e.g. `ASCTIME_BUF_MINLEN` or something like that).